packages/components/src/Textarea/Textarea.tsx

```diff
@@ -1,22 +1,58 @@
 import React from 'react';
-import { useStyles, system } from '@marigold/system';
+import { useStyles } from '@marigold/system';
+import { Exclamation, Required } from '@marigold/icons';
+import { ComponentProps } from '@marigold/types';
 
-export type TextareaProps = {
+import { ValidationMessage } from '../ValidationMessage';
+import { Label } from '../Label';
+import { Box } from '../Box';
+
+type TextareaProps = {
+  // how to check if theres a label and then htmlFor is required
   variant?: string;
-};
+  label?: string;
+  htmlFor?: string;
+  error?: string;
+  required?: boolean;
+} & ComponentProps<'textarea'>;
 
-export const Textarea = system<TextareaProps, 'textarea'>(
-  ({ variant = 'textarea', children, className, ...props }) => {
-    const classNames = useStyles(
-      {
-        variant: `form.${variant}`,
-      },
-      className
-    );
-    return (
-      <textarea className={classNames} {...props}>
+export const Textarea: React.FC<TextareaProps> = ({
+  variant = 'textarea',
+  htmlFor = 'textarea',
+  label,
+  error,
+  required = false,
+  children,
+  ...props
+}) => {
+  let errorColor = '#dd4142';
```

Comment:
why not taking it from the theme?

Reply:
still dont know how :D

Reply:
Actually you don't really need to expose the color in the component, isn't it enough to move it to the theme as it would be better if it changes with a different theme?

Reply:
@viktoria-schwarz what do you mean here?